providers/oidc.go

```diff
@@ -141,3 +168,10 @@ func (p *OIDCProvider) ValidateSessionState(s *SessionState) bool {
 
 	return true
 }
+
+func getOIDCHeader(accessToken string) http.Header {
```

Comment:
Nit: Though the Go type is `Header,` I think this actually represents a set of Headers, so I'd call this method `getOIDCHeaders`.

Looking at how this is used, are we overwriting all headers that have already been set? Would it not be better to add to the existing set?

Perhaps this method should be `addOIDCHeaders` with a signature `func addOIDCHeaders(req *http.Request)` so that we can add the Headers we care about whilst not overwriting the existing ones.

Reply:
re: naming; I was trying to stay consistent with the name of the return type.  I agree http.Header is probably a misnomer, but I wanted to be consistent.

re: overwriting; I don't think there should be any headers set already so this should be OK

Reply:
Yeah ok that makes sense